pkg/codegen/utilities.go

```diff
@@ -172,3 +172,37 @@ func ExpandShortEnumName(name string) string {
 	}
 	return name
 }
+
+// A simple in memory file system.
+type Fs struct {
```

Comment:
Why not:

```go
type Fs map[string][]byte
```

which is like `StringSet`:

https://github.com/pulumi/pulumi/blob/f33924e7aa33b2082c2721daa48632ea71c43ae4/pkg/codegen/utilities.go#L26

?

Reply:
I wen't back and forth on that. I was trying to prevent a leaky abstraction, but I don't think its possible with Go. I'll switch over to this style. 

Keeping the struct lead to copy-by-value semantics for an uninitialized `Fs` but copy by ref semantics after `Add` had been called.